code/modules/paperwork/photocopier.dm

```diff
@@ -245,6 +236,15 @@
 	flick("photocopier1", src)
 	updateUsrDialog()
 
+/obj/machinery/photocopier/proc/do_removal(obj/item/O, mob/user)
+	if(!issilicon(user)) //surprised this check didn't exist before, putting stuff in AI's hand is bad
+		O.loc = user.loc
+		user.put_in_hands(O)
+	else
+		O.loc = src.loc
+	user << "<span class='notice'>You take [O] out of [src].</span>"
+	updateUsrDialog()

```

Comment:
Are you sure this updates the window correctly? Since you nullifies copy/photocopy/doccopy after doing the window update instead of before.